pkg/http/server_group.go

```diff
@@ -0,0 +1,35 @@
+package http
+
+import (
+	"context"
+
+	"golang.org/x/sync/errgroup"
+)
+
+// NewServerGroup creates a new Server to start and gracefully stop a collection
+// of Servers.
+func NewServerGroup(servers ...Server) Server {
```

Comment:
I like this concept 👍 

On HTTP/2, gRPC - we've had a few questions on that front. Would this design open up us potentially supporting a `/oauth2/auth` handler variant for gRPC based subrequest architecture?

(I'll punt on the complexity of HTTP/2 to an upstream for now 😅 )

Reply:
Yeah I think so, as long as we can create a gRPC server that we can start/stop using the Server interface defined in server.go, I wanted to make sure the start/stop was expandable in case we need further servers in the future